.github/workflows/cicd.yaml

```diff
@@ -76,3 +76,45 @@ jobs:
       - uses: actions/checkout@v4
       - name: Test generating docs
         run: make docs
+
+  benchmark:
+    needs: [test]
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Install types
+        run: |
+          python -m pip install ./stac_fastapi/types[dev]
+
+      - name: Install extensions
+        run: |
+          python -m pip install ./stac_fastapi/extensions
+
+      - name: Install core api
+        run: |
+          python -m pip install ./stac_fastapi/api[dev,benchmark]
+
+      - name: Run Benchmark
+        run: python -m pytest stac_fastapi/api/tests/benchmarks.py --benchmark-only --benchmark-columns 'min, max, mean, median' --benchmark-json output.json
+
+      # - name: Store and benchmark result
+      #   uses: benchmark-action/github-action-benchmark@v1
+      #   with:
+      #     name: STAC FastAPI Benchmarks
+      #     tool: 'pytest'
+      #     output-file-path: output.json
+      #     alert-threshold: '130%'
+      #     comment-on-alert: true
+      #     fail-on-alert: false
+      #     # GitHub API token to make a commit comment
+      #     github-token: ${{ secrets.GITHUB_TOKEN }}
+      #     gh-pages-branch: 'gh-benchmarks'
+      #     # Make a commit only if main
+      #     auto-push: ${{ github.ref == 'refs/heads/main' }}
```

Comment:
I'll remove this comment in another commit. 

basically this will create a `gh-benchmark` branch were we will store the benchmark result. 

I'll also make a custom HTML page like https://github.com/developmentseed/tipg/blob/gh-benchmarks/dev/bench/index.html

https://developmentseed.org/tipg/benchmark.html